test/libsolidity/syntaxTests/calldata_struct_argument_with_internal_type_inside.sol

```diff
@@ -0,0 +1,9 @@
+contract C {
```

Comment:
Unless the test is really unique in some way, I think it would be better to find a subdirectory where it fits best instead of dropping it directly into `syntaxTests/`.

I'd place it next to the test from #10516.